From 834f828024ebf5361a27629219c1acd210af1a49 Mon Sep 17 00:00:00 2001
From: Esteban Gutierrez <esteban@cloudera.com>
Date: Fri, 31 Jul 2015 00:01:14 -0700
Subject: [PATCH 186/224] CLOUDERA-BUILD Revert "HBASE-13269 Limit result
 array preallocation to avoid OOME with large scan
 caching values"

Reason: New Feature
Author: Esteban Gutierrez
Ref: CDH-27345

This reverts commit 1a06908a50aefde7ada90aa89d0f1907b5f11c29.
---
 .../hadoop/hbase/regionserver/RSRpcServices.java   |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java
index 027898b..09d1f26 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java
@@ -20,7 +20,6 @@ package org.apache.hadoop.hbase.regionserver;
 
 import java.io.IOException;
 import java.io.InterruptedIOException;
-import java.lang.Math;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.net.InetSocketAddress;
@@ -2153,9 +2152,7 @@ public class RSRpcServices implements HBaseRPCErrorHandler,
           // Remove lease while its being processed in server; protects against case
           // where processing of request takes > lease expiration time.
           lease = regionServer.leases.removeLease(scannerName);
-          // Limit the initial allocation of the result array to the minimum
-          // of 'rows' or 100
-          List<Result> results = new ArrayList<Result>(Math.min(rows, 100));
+          List<Result> results = new ArrayList<Result>(rows);
           long currentScanResultSize = 0;
           long totalCellSize = 0;
 
-- 
1.7.9.5

